huxley/api/views/register.py

```diff
@@ -50,6 +59,59 @@ def register(self, request, *args, **kwargs):
             registration_serializer.is_valid(raise_exception=True)
             registration_serializer.save()
 
+        if handler is not None:
+            school_data = user_data['school']
+            address = Address(
+                line1=school_data['address'],
+                line2='',
+                city=school_data['city'],
+                country_sub_division_code=school_data['state'],
+                country=school_data['country'],
+                zip_code=school_data['zip_code']
+            )
+
+            num_delegates = sum(
+                map(
+                    int,
+                    [
+                        registration_data['num_beginner_delegates'],
+                        registration_data['num_intermediate_delegates'],
+                        registration_data['num_advanced_delegates']
+                    ]
+                )
+            )
+
+            if int(registration_data['payment_type']) == PaymentTypes.CARD:
+                payment_method = PaymentMethod.Card
+            else:
+                payment_method = PaymentMethod.Check
+
+            call_invoice_handler(
+                school_name=school_data['name'],
+                email=school_data['primary_email'],
+                phone_numbers=[school_data['primary_phone'], school_data['secondary_phone']],
+                address=address,
+                num_delegates=num_delegates,
+                payment_method=payment_method
+            )
+
         data = {'user': user_serializer.data,
                 'registration': registration_serializer.data}
         return response.Response(data, status=status.HTTP_200_OK)
+
+
+def call_invoice_handler(school_name: str,
+                         email: str,
+                         phone_numbers: List[str],
+                         address: Address,
+                         num_delegates: int,
+                         payment_method: PaymentMethod):
+    school = invoiceSchool(school_name, email, phone_numbers, address)
+    registration = invoiceRegistration(
+        school=school,
+        num_delegates=num_delegates,
+        conference=invoiceConference.BMUN71,
+        registration_date=datetime.date.today(),
+        payment_method=payment_method
+    )
+    handler.handle_registration(registration)
```

Comment:
Is there any error handling here if the module fails to make the API call to QuickBooks successfully? Registration should not fail if the API call does not go through and we should allow for requiring a manual invoice sent. On this note, it might be worth noting on registrations directly if the invoice was already sent or not so we can easily track which registrations need manual invoices.

Reply:
Sure, I'll add some error handling

Reply:
I actually have some doubts with regard to adding error handling at the moment, since we don't have proper logging implemented. Since the invoice handler is called after the registrations are serialized, as far as I'm aware, the unhandled exception won't impact functionality other than not returning a 200 response. However, it has the benefit that Django will log the exception along with a stack trace, making it easier to debug beyond knowing that invoicing failed for a particular registration.

Reply:
@srisainachuri thoughts?

Reply:
You're right in that the registration gets serialized properly, but returning a non-200 response will make it seem like it errored on the advisor end so their screen will suggest they didn't response properly. Can you try catch it to ensure that the response is still 200 if the registration was serialized properly? Additionally, for invoices tracking purposes, it might be worthfile to add a checkbox to the registration model if invoices were sent or not, so you can easily manually send the ones that need to be sent instead of parsing through django stack traces

Reply:
Here's the [registration handler](https://github.com/bmun/huxley/blob/cbd2c2284ae1c1e7275c1855586647fd12f70951/huxley/www/js/components/RegistrationView.js#L545) in the frontend: